src/dmd/statementsem.d

```diff
@@ -2683,22 +2683,35 @@ private extern (C++) final class StatementSemanticVisitor : Visitor
                 ed = ds.isEnumDeclaration();
             if (ed)
             {
-              Lmembers:
+                int missingMembers = 0;
+                const maxShown = !global.params.verbose ? 6 : int.max;
+            Lmembers:
                 foreach (es; *ed.members)
                 {
                     EnumMember em = es.isEnumMember();
                     if (em)
                     {
                         foreach (cs; *ss.cases)
                         {
-                            if (cs.exp.equals(em.value) || (!cs.exp.type.isString() && !em.value.type.isString() && cs.exp.toInteger() == em.value.toInteger()))
+                            if (cs.exp.equals(em.value) || (!cs.exp.type.isString() &&
+                                !em.value.type.isString() && cs.exp.toInteger() == em.value.toInteger()))
                                 continue Lmembers;
                         }
-                        ss.error("`enum` member `%s` not represented in `final switch`", em.toChars());
-                        sc.pop();
-                        return setError();
+                        if (missingMembers == 0)
+                            ss.error("missing cases for `enum` members in `final switch`:");
+
+                        if (missingMembers < maxShown)
+                            ss.errorSupplemental("`%s`", em.toChars());
+                        missingMembers++;
                     }
                 }
+                if (missingMembers > 0)
+                {
+                    if (missingMembers > maxShown)
+                        ss.errorSupplemental("... (%d more, -v to show) ...", missingMembers - maxShown);
```

Comment:
```suggestion
                        errorSupplemental(ss.loc, "... (%d more, -v to show) ...", missingMembers - maxShown);
```
There's nothing really gained from adding a new method.

Reply:
Why do `Statement.error` and `Expression.errorSupplemental` exist then? My guess was it is structured this way to make it easier to replace with something `pure` later on or something, but I don't know what the idea behind error member functions is.

Reply:
The current error reporting system is archaic and should really be reworked to be cleaner and more flexible.

However, this is a different discussion. I suggest focusing on the minimal required changes for this PR.

Reply:
I removed Statement.errorSupplemental now